infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -188,7 +191,15 @@ public ServerContext(Map<String, Object> serverConfig) {
         // Metrics setup & reporting configuration
         if (!isMetricsReportingSetUp(metrics)) {
             MetricsUtils.metricsReportingSetup(metrics);
+            logMetricsSize();
         }
+
+        FileChannelPerf.setupParameters(OP_SPIKE_CNT, OP_SPIKE_LATENCY, false);
```

Comment:
Why this is a static method? Can we make it non-static? 
If we make it non static we improve testability and make it more flexible

Reply:
One jvm has one stats measurements. Each filechannel has a corresponding filechannelperf.

Reply:
It makes code much more fragile, please make all the fields in `FileChannelPerf` non-static and pass an instance of FileChannelPerf class into ServerContext constructor. Or any other way that works, in this case, like create an instance of FileChannelPerf inside ServerContext constructor.
Reading about `dependecy injection` mechanism and how it works in Java will be a good help and provides insight why you must not use static in these cases and why DI is important.
In addition information about SOLID patters will be also helpful

Reply:
These values should be configurable. The way configurations are passed to the servers is through the serverContext options and https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java#L44